api/v1alpha1/hosted_controlplane.go

```diff
@@ -37,10 +37,20 @@ type HostedControlPlaneSpec struct {
 type HostedControlPlaneStatus struct {
 	// Ready denotes that the HostedControlPlane API Server is ready to
 	// receive requests
+	// +kubebuilder:validation:Required
 	// +kubebuilder:default=false
 	Ready bool `json:"ready"`
 
+	// ControlPlaneEndpoint contains the endpoint information by which
+	// external clients can access the control plane.  This is populated
+	// after the infrastructure is ready.
+	// +kubebuilder:validation:Optional
 	ControlPlaneEndpoint APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// +kubebuilder:validation:Required
+	// Condition contains details for one aspect of the current state of the HostedControlPlane.
+	// Current condition types are: "Available"
+	Conditions []metav1.Condition `json:"conditions"`
```

Comment:
Should we have a HostedControlPlaneCondition type so we can have an enum for Type? (See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)